src/js/actions/deviceActions.js

```diff
@@ -590,6 +592,28 @@ export const getSessionDetails = (sessionId, deviceId, userId, startDate, endDat
   );
 };
 
+export const deviceFileDownload = (deviceId, path) => () => GeneralApi.post(`${deviceConnect}/devices/${deviceId}/download`, { path });
```

Comment:
Download technically works (I can see the response with the dev tools), but nothing happens in the UI and the file is not downloaded. This is tricky, I would try to let the browser handle the download of the content-disposition: attachment file, but to do that you cannot rely on XHR requests. The only (hackish?) solution I can think of is creating a hidden iframe with the post form, target a new window and submit it, it should trigger the download). Another option would be to change the backend and implement the download as a GET end-point, passing the params as GET parameters and relying on the Cookie to pass in the JWT token. What do you think?